test/e2e-sno-disruptive/sno_disruptive_test.go

```diff
@@ -59,10 +59,11 @@ func waitForClusterInGoodState(t testing.TB, cs clientSet, waitPollTimeout, must
 	t.Helper()
 
 	startTs := time.Now()
-	t.Logf("Waiting %s for the cluster to be in a good condition, interval = 10s, timeout %v", mustBeReadyFor.String(), waitPollTimeout)
+	t.Logf("Waiting %s for the cluster to be in a good condition, interval = 20s, timeout %v", mustBeReadyFor.String(), waitPollTimeout)
 
-	return wait.Poll(10*time.Second, waitPollTimeout, func() (bool, error) {
+	return wait.PollUntilContextTimeout(context.Background(), 20*time.Second, waitPollTimeout, true, func(cxt context.Context) (bool, error) {
```

Comment:
could use update the signature of the `waitForClusterInGoodState` function to accept a context instead ?
```
waitForClusterInGoodState(ctx context.Context, t testing.TB, cs clientSet, waitPollTimeout, mustBeReadyFor time.Duration) error
```

Reply:
Updated.